api-examples/pretrain_paired_pytorch.py

```diff
@@ -251,9 +254,16 @@ def train():
             batch = next(train_itr)
             steps += 1
             x, y = batch
-            inputs = x.to(args.device)
-            labels = y.to(args.device)
-            loss = loss_function(inputs, labels)
+            if args.model_type == 'encoder-decoder':
```

Comment:
it might be cleaner to bind a function pointer like `run_step` to `run_step_s2s` vs `run_step_dual` at initialization time based on model_type

Reply:
done